File: tests/human/x/pl/match_full.pl
:- initialization(main).
classify(0,'zero').
classify(1,'one').
classify(_, 'many').
main :-
    X=2,
    (X=:=1->L1='one'; X=:=2->L1='two'; X=:=3->L1='three'; L1='unknown'),
    writeln(L1),
    Day='sun',
    (Day='mon'->Mood='tired'; Day='fri'->Mood='excited'; Day='sun'->Mood='relaxed'; Mood='normal'),
    writeln(Mood),
    Ok=true,
    (Ok=true->Status='confirmed'; Status='denied'),
    writeln(Status),
    classify(0,R0), writeln(R0),
    classify(5,R1), writeln(R1),
    halt.

File: tests/human/x/clj/match_full.clj
(ns match-full)

(let [x 2
      label (cond
              (= x 1) "one"
              (= x 2) "two"
              (= x 3) "three"
              :else "unknown")]
  (println label))

(let [day "sun"
      mood (cond
             (= day "mon") "tired"
             (= day "fri") "excited"
             (= day "sun") "relaxed"
             :else "normal")]
  (println mood))

(let [ok true
      status (if ok "confirmed" "denied")]
  (println status))

(defn classify [n]
  (cond
    (= n 0) "zero"
    (= n 1) "one"
    :else "many"))

(println (classify 0))
(println (classify 5))
